tests/test_observers/test_file_storage_observer.py

```diff
@@ -117,11 +117,21 @@ def test_fs_observer_started_event_creates_rundir(dir_obs, sample_run, monkeypat
     def mkdir_raises_file_exists(name, mode=0o777):
         raise FileExistsError("File already exists: " + name)
 
+    # Assume some problem with the filesystem exists
```

Comment:
I would break up this test case into three separate cases, as they test different behavior. Having to document code blocks with comments is usually a sign that things should be broken up into smaller pieces ;)

Reply:
Done